### PR TITLE
wrap <64bit types properly in loop unrolling

### PR DESCRIPTION
**Pull request by ChrisDodd**

### CODE REVIEW

**Comment by jafingerhut** [comment]
Is there a test P4 program you can add that works with these changes, but gives wrong results without these changes?

**Comment by ChrisDodd** [comment]
Looks like the bazel build is screwed up again

**Comment by fruffy** [comment]
@smolkaj The Bazel cache is acting up a lot recently. Maybe we should change the way it works for our CI?

**Comment by ChrisDodd** [comment]
> Is there a test P4 program you can add that works with these changes, but gives wrong results without these changes?

I added your tricky2 testcase, since that one needs unrolling to actually work, not just reject it.

**Review by jafingerhut — APPROVED** [review]
The unrolled version of the test program in the -midend.p4 expected output file looks correct to me.  I have not attempted to review in detail the C++ code changes (sorry).

**Comment by jafingerhut** [comment]
Looks like the test program needs a .txtpb expected P4Info output file, too.

**Comment by fruffy** [comment]
> Looks like the bazel build is screwed up again

I cleared the cache and tests are passing now.

File: testdata/p4_16_samples/loop-3-clause-tricky2.p4 [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+control ingressImpl(inout headers_t hdr,
+                    inout metadata_t meta,
+                    inout standard_metadata_t stdmeta)
+{
+    bit<8> n = hdr.ethernet.srcAddr[15:8];
+    bit<3> i;
+    apply {
+        // Correct sequence of values that i iterates through: 0, 3,
+        // 1, 4
```
Comment (jafingerhut) [inline testdata/p4_16_samples/loop-3-clause-tricky2.p4:52]: This is my mistake, but if you wouldn't mind fixing the comment so the list of values i iterates through is correct: 0, 3, 6, 1, 4, that would be appreciated.

**Pull request merged**